app/src/main/java/com/example/weather/MainActivity.java

```diff
@@ -2,13 +2,173 @@
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import javax.net.ssl.HttpsURLConnection;
 
 public class MainActivity extends AppCompatActivity {
 
+    private TextView msgTemperature;
+    private EditText msgCity;
+    private Button buttonByCity;
+    private Button buttonByLocation;
+
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        msgTemperature = findViewById(R.id.msg_temperature);
+        msgCity = findViewById(R.id.msg_city);
+        buttonByCity = findViewById(R.id.button_by_city);
+        buttonByLocation = findViewById(R.id.button_by_location);
+    }
+
+    public void onClickByCity(View view) {
+        Log.d("myTag", "OnClick start");
+        String city = msgCity.getText().toString();
+        if (!checkCityString(city)) {
+            msgOnWrongCity();
+            return;
+        }
+
+        String request = formatRequestWithCity(city);
+        Log.d("myTag", request);
+        String response = doRequest(request);
+        if (response != null) {
+            Log.d("myTag", "len:" + response.length());
+        } else {
+            msgOnWrongRequest();
+            return;
+        }
+
+        String formattedOutput = getTemperature(response);
+        if (formattedOutput == null) {
+            msgOnWrongCity();
+            return;
+        }
+        String newMessage = getText(R.string.default_message) + " " + formattedOutput + " \u00B0C";
+        msgTemperature.setText(newMessage);
+        Log.d("myTag", "OnClick finish");
+    }
+
+    public void onClickByLocation(View view) {
+
+    }
+
+    private boolean checkCityString(String city) {
+        if (city == null) {
+            msgOnWrongCity();
+            return false;
+        }
+        else if (city.length() < 2 || city.length() > 30) {
+            msgOnWrongCity();
+            return false;
+        }
+        // TODO: do some other checking of input string
+        return true;
+    }
+
+    private String formatRequestWithCity(String city) {
+        String api_url = getText(R.string.weather_entry_point).toString();
+        String key = getText(R.string.weather_key).toString();
+        return api_url + "?q=" + city + "&appid=" + key + "&units=metric";
+    }
+
+    private String getTemperature(String response) {
+        String ret = null;
+
+        try {
+            JSONObject json = new JSONObject(response);
+            JSONObject main  = json.getJSONObject("main");
+            ret = main.getString("temp");
+            Log.d("myTag", "temperature = " + ret);
+        } catch (JSONException e) {
+            msgOnWrongRequest();
+            e.printStackTrace();
+        }
+
+        return ret;
+    }
+
+    private String doRequest(String request) {
+        final String[] respond = {null};
```

Comment:
respond variable always has only 0 or 1 item. No need to have it as an array.
```suggestion
        final String respond = null;
```